OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -266,13 +266,16 @@ public void UpdateMouseover()
 				return;
 			}
 
-			if (resourceRenderer != null)
+			if (resourceRenderers.Any())
 			{
-				var resource = resourceRenderer.GetRenderedResourceType(cell);
-				if (resource != null)
+				foreach (var resourceRenderer in resourceRenderers)
 				{
-					TooltipType = WorldTooltipType.Resource;
-					ResourceTooltip = resource;
+					var resource = resourceRenderer.GetRenderedResourceType(cell);
+					if (resource != null)
+					{
+						TooltipType = WorldTooltipType.Resource;
+						ResourceTooltip = resource;
+					}
```

Comment:
In my opinion we should break from the `foreach` loop as soon as we find a match and populated the tooltip, to avoid needlessly checking other resource renderers. It's extremely unlikely there will be any other resource on the same cell anyway, and even if there were, taking the first instead of the last entry is still preferrable if we can save additional look-ups that way.

Reply:
I agree. Done.

Reply:
The map resource + custom terraintype layers can only contain one entry, so this will be the least of the problems if a mod does try to have multiple resources in a cell.